kivy/app.py

```diff
@@ -654,7 +661,10 @@ def get_application_config(self):
         '''
 
         if platform == 'android':
-            defaultpath = '/sdcard/.%(appname)s.ini'
+            defaultpath = join(
+                dirname(abspath(sys.executable)),
```

Comment:
Isn't `sys.executable` just `''` with python2 or `android_python` on Android? It would probably be better to just check the environment variables (`ANDROID_PRIVATE`) etc. to find the app's own directory.